### PR TITLE
Add project geofencing settings

### PR DESCRIPTION
**Pull request by nirvn**
A couple of project-level settings needed to implement geofencing in qfieldsync / QField.

### CODE REVIEW

File: libqfieldsync/project.py [hunk 1]
```diff
@@ -105,6 +119,58 @@ def digitizing_logs_layer(self, value):
...
+    @geofencing_behavior.setter
+    def geofencing_behavior(self, value):
+        self.project.writeEntry(
+            "qfieldsync", ProjectProperties.GEOFENCING_BEHAVIOR, value
+        )
+
+    @property
+    def geofencing_prevent_digitizing(self):
+        geofencing_prevent_digitizing, _ = self.project.readBoolEntry(
+            "qfieldsync", ProjectProperties.GEOFENCING_PREVENT_DIGITIZING, False
```
Comment (suricactus) [inline libqfieldsync/project.py:151]:
```suggestion
            "qfieldsync", ProjectProperties.GEOFENCING_SHOULD_PREVENT_DIGITIZING, False
```

File: libqfieldsync/project.py [hunk 2]
```diff
@@ -105,6 +119,58 @@ def digitizing_logs_layer(self, value):
...
+    @geofencing_prevent_digitizing.setter
+    def geofencing_prevent_digitizing(self, value):
+        self.project.writeEntry(
+            "qfieldsync", ProjectProperties.GEOFENCING_PREVENT_DIGITIZING, value
+        )
+
+    @property
+    def geofencing_active(self):
+        geofencing_active, _ = self.project.readBoolEntry(
+            "qfieldsync", ProjectProperties.GEOFENCING_ACTIVE, False
```
Comment (suricactus) [inline libqfieldsync/project.py:164]:
```suggestion
            "qfieldsync", ProjectProperties.GEOFENCING_IS_ACTIVE, False
```

**Review by suricactus — COMMENTED** [review]
Hey, all looks good, just the classic comment about boolean config.

**Comment by nirvn** [comment]
@suricactus , addressed.

**Pull request merged**